packages/fxa-settings/src/pages/ConnectAnotherDevice/index.tsx

```diff
@@ -0,0 +1,249 @@
+import { RouteComponentProps, Link, navigate } from '@reach/router';
+import { usePageViewEvent } from '../../lib/metrics';
+import { ENTRYPOINTS } from '../../constants';
+import { HeartsVerifiedImage } from '../../components/images';
+import Banner, { BannerType } from '../../components/Banner';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { REACT_ENTRYPOINT } from '../../constants';
+
+export type ConnectAnotherDeviceProps = {
+  email: string;
+  forceView?: boolean;
+  entrypoint: ENTRYPOINTS; // in Backbone, this came in through the queryparam 'entrypoint' and would be validated in the component, defaulting to FXA Menu entrypoint
+  isSignedIn: boolean; // check if `isSignedInAccount` (on "account"), or get `isVerificationSameBrowser` on the broker
+  showSuccessMessage?: boolean; // check if you have a truthy value for "showSuccessMessage" either on the model, or in a search param by the same name. This is so that if you hit this url, you don't automatically be an inaccurate status message
+  isSignUp: boolean; // we get this by getting the verification reason (or "type") off of the model
+  isSignIn: boolean; // we get this by getting the verification reason (or "type") off of the model
+  canSignIn: boolean; //we check if the user is 1. not signed in, and 2. on a UA that supports sync auth (i.e., Firefox Desktop where version > 40, or Firefox android where version > 43)
+  device?: Devices;
+};
+
+export enum Devices {
+  FIREFOX_MENU = 'Firefox Menu',
+  FIREFOX_ANDROID = 'Firefox Android',
+  FIREFOX_DESKTOP = 'Firefox Desktop',
+  FIREFOX_IOS = 'Firefox Ios',
+  OTHER_ANDROID = 'Other Android',
+  OTHER_IOS = 'Other Ios',
+  OTHER = 'Other',
+}
+export const viewName = 'connect-another-device';
+const ConnectAnotherDevice = ({
+  email,
+  forceView = false,
+  entrypoint,
+  isSignedIn,
+  showSuccessMessage = false,
+  isSignUp,
+  isSignIn,
+  canSignIn,
+  device = Devices.FIREFOX_MENU,
+}: ConnectAnotherDeviceProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+  const getEscapedSignInUrl = () => {
+    // TODO:
+    // recreate the escaped signin url.
+    // basically this means pulling a bunch of params off of the relier
+    // you can see this in the `sync-auth-mixin.js`, and in `connect_another_device.js`
+    return '/signin';
+  };
+
+  const isEligibleForPairing = () => {
+    // TODO:
+    // get context from `relier.get('context')`
+    // originally we got the entrypoint from the relier, but here we have it as a prop
+    // get the action from the relier as well
+    // if:
+    //    the user is signed into the default account,
+    //    AND the context === FX_DESKTO_V3_CONTEXT
+    //    AND (
+    //       EITHER the entrypoint === FIREFOX_TOOLBAR_ENTRYPOINT)
+    //       OR (the action !== email, and the entrypoint is firefox preferences || firefox synced tabs || firefox tabs sidebar || firefox menu || firefox fx view )
+    // then:
+    const isEligible = false;
```

Comment:
Won't this always return `false`? Maybe just a comment in this method to explain what makes a device eligible/not-eligible until we add in functionality with the relier?

Reply:
Oh I thought I had indicated that with the above comment but clearly I did not!! I'll try to make it clearer!

Reply:
Oh, I see what you mean. I was wondering if we should temporarily pass in the `isEligible` value as a prop, but I suppose that won't make any difference for Storybook as the redirect wouldn't show up there.